connect/runtime/src/test/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStoreTest.java

```diff
@@ -94,8 +93,8 @@ public void testFlushFailureWhenWriteToSecondaryStoreFailsForTombstoneOffsets()
         AtomicReference<Throwable> callbackError = new AtomicReference<>();
 
         Future<Void> setFuture = offsetBackingStore.set(getSerialisedOffsets(mkMap(
-            mkEntry(OFFSET_KEY_SERIALIZED, null),
```

Comment:
this case needs to have map carrying null value, so you can add helper for this case specifically

Reply:
Since there are still many tests using mkEntry to create entries with null values, this PR will migrate all the code that can be converted to use Map.entry. The remaining code that still uses mkEntry will only for creating null entries.